contracts/manager/DelegatedManager.sol

```diff
@@ -0,0 +1,371 @@
+
+    /**
+     * Throws if the sender is not the SetToken methodologist
+     */
+    modifier onlyMethodologist() {
+        require(msg.sender == methodologist, "Must be methodologist");
+        _;
+    }
+
+    /**
+     * Throws if the sender is not a listed extension
+     */
+    modifier onlyExtension() {
+        require(extensionAllowlist[msg.sender] == ExtensionState.INITIALIZED, "Must be initialized extension");
+        _;
+    }
+
+    /* ============ State Variables ============ */
+
+    // Instance of SetToken
+    ISetToken public immutable setToken;
+
+    // Address of factory contract used to deploy contract
+    address public immutable factory;
+
+    // Mapping to check if extension is enabled
+    mapping(address => ExtensionState) public extensionAllowlist;
+
+    // Array of enabled extensions
+    address[] internal extensions;
+
+    // Mapping indicating if address is an approved operator
+    mapping(address=>bool) public operatorAllowlist;
+
+    // List of approved operators
+    address[] internal operators;
+
+    // Mapping indicating if asset is approved to be traded for, wrapped into, claimed, etc.
+    mapping(address=>bool) public assetAllowlist;
+
+    // List of allowed assets
+    address[] internal allowedAssets;
+
+    // Address of methodologist which serves as providing methodology for the index
+    address public methodologist;
+
+    /* ============ Constructor ============ */
+
+    constructor(
+        ISetToken _setToken,
+        address _factory,
+        address _methodologist,
+        address[] memory _extensions,
+        address[] memory _operators,
+        address[] memory _allowedAssets
+    )
+        public
+    {
+        setToken = _setToken;
+        methodologist = _methodologist;
+        factory = _factory;
+
+        _addExtensions(_extensions);
+        _addOperators(_operators);
+        _addAllowedAssets(_allowedAssets);
+    }
+
+    /* ============ External Functions ============ */
+
+    /**
+     * ONLY EXTENSION: Interact with a module registered on the SetToken.
+     *
+     * @param _module           Module to interact with
+     * @param _data             Byte data of function to call in module
+     */
+    function interactManager(address _module, bytes calldata _data) external onlyExtension {
+        // Invoke call to module, assume value will always be 0
+        _module.functionCallWithValue(_data, 0);
+    }
+
+    /**
+     * Initializes an added extension from PENDING to INITIALIZED state. An address can only
+     * enter a PENDING state if it is an enabled module added by the manager. Only callable
+     * by the module itself, hence msg.sender is the subject of update.
+     */
+    function initializeExtension() external {
+        require(extensionAllowlist[msg.sender] == ExtensionState.PENDING, "Extension must be pending");
+
+        extensionAllowlist[msg.sender] = ExtensionState.INITIALIZED;
+        extensions.push(msg.sender);
+    }
```

Comment:
do we want to emit an ExtensionInitialized(msg.sender) event here?

Reply:
+1